packages/platform-sdk-dot-indexer/src/index.ts

```diff
@@ -0,0 +1,73 @@
+import PQueue from "p-queue";
+import retry from "p-retry";
+import pino from "pino";
+
+import { indexBlock, useElasticSearch, usePolkadot } from "./helpers";
+
+export const subscribe = async (flags: Record<string, string>): Promise<void> => {
+	// Logging
+	const logger = pino({ name: "@arkecosystem/platform-sdk-dot-indexer", level: "debug" });
+
+	// Queue
+	const queue = new PQueue({ autoStart: false, concurrency: 10 });
+	// queue.on("active", () => logger.debug(`Size: ${queue.size}  Pending: ${queue.pending}`));
+	// queue.on("idle", () => logger.debug(`Queue is idle. Size: ${queue.size} | Pending: ${queue.pending}`));
+	// queue.on("add", () => logger.debug(`Task is added. Size: ${queue.size} | Pending: ${queue.pending}`));
+	// queue.on("next", () => logger.debug(`Task is completed. Size: ${queue.size} | Pending: ${queue.pending}`));
+
+	// Storage
+	const elastic = useElasticSearch(flags.elasticsearch);
+
+	// API
+	let polkadot = await usePolkadot(flags.polkadot);
+
+	// @TODO: get last block stored in elasticsearch and set `i` to that value to prevent full reindexing
+	const latestBlockHeight = parseInt((await polkadot.derive.chain.bestNumberFinalized()).toString());
+
+	for (let i = 1; i < latestBlockHeight; i++) {
+		try {
+			if (queue.size === 250) {
+				logger.info("Draining Queue...");
+
+				await queue.start().onIdle();
+				queue.pause();
+
+				logger.info("Drained Queue...");
+			}
+
+			// @ts-ignore
+			// eslint-disable-next-line @typescript-eslint/no-floating-promises
+			queue.add(() =>
+				retry(() => indexBlock(i, polkadot, elastic, logger), {
+					onFailedAttempt: (error) => {
+						console.log(error);
+						logger.error(
+							`Attempt ${error.attemptNumber} failed. There are ${error.retriesLeft} retries left.`,
+						);
+					},
+					retries: 5,
+				}),
+			);
+		} catch (error) {
+			console.log(error);
+
+			// Connections can randomly be dropped so we should try to reconnect but
+			// also terminate when the connection drops too often which means a human
+			// should probably take a look at this or we should halt polling of data.
+
+			logger.info("Disconnecting...");
+
+			await polkadot.disconnect();
+
+			logger.info("Reconnecting...");
+
+			polkadot = await usePolkadot(flags.polkadot);
+
+			// Reduce the count to go back a block before we continue to loop
+			i--;
```

Comment:
:skull: This is dangerous, a potential for halting in the loop.

Reply:
Yeah this is just temporary because I am still thinking on how to best handle a dropped connection. I might re-queue the failed job and just keep going.